sdk/search/azure-search-documents/tests/async_tests/test_search_client_basic_live_async.py

```diff
@@ -3,79 +3,43 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-import asyncio
-import functools
-import json
-from os.path import dirname, join, realpath
 
 import pytest
-
-from devtools_testutils import AzureMgmtTestCase, ResourceGroupPreparer
-from azure_devtools.scenario_tests import ReplayableTest
-
-from search_service_preparer import SearchServicePreparer
-
-from azure_devtools.scenario_tests.utilities import trim_kwargs_from_test_function
-
-CWD = dirname(realpath(__file__))
-
-SCHEMA = open(join(CWD, "..", "hotel_schema.json")).read()
-BATCH = json.load(open(join(CWD, "..", "hotel_small.json"), encoding='utf-8'))
-
 from azure.core.exceptions import HttpResponseError
-from azure.core.credentials import AzureKeyCredential
 from azure.search.documents.aio import SearchClient
+from devtools_testutils.aio import recorded_by_proxy_async
+from devtools_testutils import AzureRecordedTestCase
 
-TIME_TO_SLEEP = 3
-
-def await_prepared_test(test_fn):
-    """Synchronous wrapper for async test methods. Used to avoid making changes
-    upstream to AbstractPreparer (which doesn't await the functions it wraps)
-    """
-
-    @functools.wraps(test_fn)
-    def run(test_class_instance, *args, **kwargs):
-        trim_kwargs_from_test_function(test_fn, kwargs)
-        loop = asyncio.get_event_loop()
-        return loop.run_until_complete(test_fn(test_class_instance, **kwargs))
-
-    return run
-
+from search_service_preparer import SearchEnvVarPreparer, search_decorator
 
-class SearchClientTestAsync(AzureMgmtTestCase):
-    FILTER_HEADERS = ReplayableTest.FILTER_HEADERS + ['api-key']
+class TestSearchClientAsync(AzureRecordedTestCase):
 
-    @ResourceGroupPreparer(random_name_enabled=True)
-    @SearchServicePreparer(schema=SCHEMA, index_batch=BATCH)
-    async def test_async_get_document_count(
-        self, api_key, endpoint, index_name, **kwargs
-    ):
-        client = SearchClient(
-            endpoint, index_name, AzureKeyCredential(api_key)
-        )
+    @SearchEnvVarPreparer()
+    @search_decorator(schema="hotel_schema.json", index_batch="hotel_small.json")
+    @recorded_by_proxy_async
+    async def test_get_document_count(self, endpoint, api_key, index_name):
+        client = SearchClient(endpoint, index_name, api_key)
         async with client:
             assert await client.get_document_count() == 10
 
-    @ResourceGroupPreparer(random_name_enabled=True)
-    @SearchServicePreparer(schema=SCHEMA, index_batch=BATCH)
-    async def test_get_document(self, api_key, endpoint, index_name, **kwargs):
-        client = SearchClient(
-            endpoint, index_name, AzureKeyCredential(api_key)
-        )
+    @SearchEnvVarPreparer()
+    @search_decorator(schema="hotel_schema.json", index_batch="hotel_small.json")
```

Comment:
curious, did Pytest have a warning around the test being async but not being decorated with the async marker due to strict mode being enabled in the future?

Reply:
Nope, but some of the unit tests do complain about "async def function and no async plugin installed".

Reply:
A quick pip install in my venv and those tests pass too--no warnings.

Reply:
![image](https://user-images.githubusercontent.com/5723682/151844340-a54ce3e1-0ad4-441c-b058-07dbc85a5eb1.png)